iceoryx_hoofs/include/iceoryx_hoofs/cxx/requires.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_CXX_REQUIRES_HPP
+#define IOX_HOOFS_CXX_REQUIRES_HPP
+
+#include "iceoryx_hoofs/platform/platform_correction.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+namespace internal
+{
+void Require(
+    const bool condition, const char* file, const int line, const char* function, const char* conditionString) noexcept;
+} // namespace internal
+
+// implementing C++ Core Guideline, I.6. Prefer Expects
+// see:
+// https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-expects
+#define Expects(condition) internal::Require(condition, __FILE__, __LINE__, __PRETTY_FUNCTION__, #condition)
```

Comment:
It is really unfortunate that macros are not namespaced. While this macro will produce an error when used without the `iox` namespace, it can still cause issues when someone else also has an `Expects` macro. It is really a pity that the core guidelines do not recommend to prefix this macro in order to prevent collisions.

It is nothing for the PR, but do you think it would be worth to rename this to `IOX_EXPECTS`?

Reply:
Let's wait how the `Ensures`, `Expects` or `Requires` discussion ends and then we may have only an `IOX_REQUIRES` or something completely different.